File: sevent_ lesson_intermediate_certifications/C_Sharp_ok/ConsoleApp1/Person.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ConsoleApp1
{
    public abstract class Person : IParent, IBabySitter
    {
        public readonly string Name;
        public readonly DateTime Birthday;
        protected abstract string HelloPhrase { get; set; }

        public Person(string name, DateTime birthday)
        {
            Birthday = birthday;
            Name = name;
        }

        public Person(string name)
        {
            Name = name;
            Birthday = DateTime.Now;
        }

        public virtual void SayHello()
        {
            Console.WriteLine($"{Name}: Привет, я человек!");
        }

        public virtual void SayHelloPhrase()
        {
            Console.WriteLine($"{Name}: {HelloPhrase}");
        }

        public void Print()
        {
            Console.WriteLine($"Имя - {Name}; День рождения - {Birthday.ToString("dd.MM.yyyy")}");
        }

        protected abstract void TakeCareImlementation();

        public void TakeCare()
        {
            TakeCareImlementation();
        }
    }
}

File: sevent_ lesson_intermediate_certifications/Java_ok/Person.java
import java.time.LocalDate;

public abstract class Person
{
    public String name;
    public LocalDate birthday;
    protected String helloPhrase;
    
    public Person(String name)
    {
        this.name = name;
    }
    public Person(String name, LocalDate birthday)
    {
        this.birthday = birthday;
        this.name = name;
    }

    public void SayHello()
    {
        System.out.printf("%s: Привет, я человек!\n", name);
    }

    public void SayHelloPhrase()
    {
        System.out.printf("%s: %s \n", name, helloPhrase);
    }

    public void Print()
    {
        System.out.printf("Имя = %s, день рождения = %s\n", name, birthday);
    }

    protected abstract void TakeCareImlementation();

    public void TakeCare()
    {
        TakeCareImlementation();
    }
}
